base/array.jl

```diff
@@ -1618,8 +1618,14 @@ julia> findnext(A,3)
 function findnext(A, start::Integer)
     l = endof(A)
     i = start
+    warned = false
     while i <= l
-        if A[i] != 0
+        a = A[i]
+        if !warned && !(a isa Bool)
+            depwarn("In the future `findnext` will only work on boolean collections. Use `findnext(x->x!=0, A)` instead.", :findnext)
```

Comment:
Should `start` be an argument to `findnext` in the depwarn message?
```
depwarn(" ... Use `findnext(x->x!=0, A, start)` instead.", :findnext)
```

Similarly for `findprev`

Reply:
Indeed, it should.